rasa/shared/core/domain.py

```diff
@@ -1446,3 +1450,94 @@ def slot_mapping_for_form(self, form_name: Text) -> Dict[Text, Any]:
             The slot mapping or an empty dictionary in case no mapping was found.
         """
         return self.forms.get(form_name, {})
+
+
+class SlotMapping(Enum):
+    """Defines the available slot mappings."""
+
+    FROM_ENTITY = 0
+    FROM_INTENT = 1
+    FROM_TRIGGER_INTENT = 2
+    FROM_TEXT = 3
+
+    def __str__(self) -> Text:
+        """Returns a string representation of the object."""
+        return self.name.lower()
+
+    @staticmethod
+    def validate(mapping: Any, form_name: Text, slot_name: Text) -> None:
```

Comment:
Should the type of `mapping` be `Dict`?

Reply:
We don't know at this point. That's why I used `Any`.

Reply:
But if we throw an error when it is not `Dict` then shouldn't we have the type here be `Dict` so potential errors can be caught using static analysis?